### PR TITLE
8 concept activation collapse

### PR DESCRIPTION
**Pull request by PortikAbel**

### CODE REVIEW

File: src/scripts/run.py [hunk 1]
```diff
@@ -14,7 +14,7 @@
 load_dotenv()
 
 
-@hydra.main(version_base=None, config_path="../config/yaml", config_name="config")
+@hydra.main(version_base=None, config_path="../../config/hydra", config_name="config")
```
[the remark this answers is not in the document]
Comment (PortikAbel) [reply src/scripts/run.py:17]: config is gitignored

File: src/model/model.py [hunk 2]
```diff
@@ -47,45 +39,24 @@ def __init__(self, cfg: ReProSegConfig, log: Log):
...
+            checkpoint_path=cfg.model.backbone_checkpoint,
         )
-        self.feature_net = features
+        self.backbone = features
         self.aspp_convs = aspp_convs
 
         self.shared_weights = self.aspp_convs[0][0].weight
         # set shared weights to all aspp convolutions
         for conv in self.aspp_convs:
             conv[0].weight.data = self.shared_weights
```
[the remark this answers is not in the document]
Comment (PortikAbel) [reply src/model/model.py:51]: will be addressed in https://github.com/PortikAbel/ReProSeg/issues/10

File: src/train/loss.py [hunk 3]
```diff
@@ -54,7 +54,7 @@ def calculate_loss(
         loss += weights.tanh * tanh_loss
 
     if train_phase is not TrainPhase.PRETRAIN:
-        softmax_inputs = torch.log1p(out**2)
+        softmax_inputs = torch.nn.functional.softmax(torch.log1p(out**2), dim=1)
         class_loss = criterion(softmax_inputs, ys.squeeze())
```
[the remark this answers is not in the document]
Comment (PortikAbel) [reply src/train/loss.py:59]: Will be addressed in https://github.com/PortikAbel/ReProSeg/issues/11

File: src/model/model.py [hunk 4]
```diff
@@ -316,9 +213,9 @@ def __init__(
...
 
     def forward(self, input_: torch.Tensor) -> torch.Tensor:
-        weight = torch.where(self.weight < self.MIN_CLASSIFICATION_WEIGHT, 0.0, self.weight)
+        weight = F.softplus(self.weight)
         return F.conv2d(input_, weight, self.bias, stride=1, padding=0)
 
     @property
-    def used_prototypes(self) -> torch.Tensor:
+    def used_concepts(self) -> torch.Tensor:
         return (self.weight >= self.MIN_CLASSIFICATION_WEIGHT).any(dim=0).squeeze().nonzero().squeeze()
```
[the remark this answers is not in the document]
Comment (PortikAbel) [reply src/model/model.py:221]: It is intentional

File: src/model/model.py [hunk 5]
```diff
@@ -160,136 +131,62 @@ def _init_classifier_weights(self):
...
-                else:
-                    param.requires_grad = False
-        else:
-            self._log.warning("Layer groups not implemented for selected backbone architecture.")
+        self.param_groups["classifier_head"] = []
+
+        for _name, param in self.layers.backbone.named_parameters():
+            self.param_groups["backbone"].append(param)
+
+        self.param_groups["classifier_head"].append(self.layers.shared_weights)
```
[the remark this answers is not in the document]
Comment (PortikAbel) [reply src/model/model.py:141]: Will be addressed by https://github.com/PortikAbel/ReProSeg/issues/12

File: src/model/model.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (PortikAbel) [reply src/model/model.py:221]: Will be addressed by https://github.com/PortikAbel/ReProSeg/issues/13

File: src/visualize/visualizer.py [hunk 6]
```diff
@@ -154,14 +154,14 @@ def render_prototype_activations(self):
             grid = torchvision.utils.make_grid(all_tensors, nrow=self.k + 1, padding=1)
             torchvision.utils.save_image(grid, self.log.prototypes_dir / f"grid_top_{self.k}_prototype_activations.png")
```
[the remark this answers is not in the document]
Comment (PortikAbel) [reply src/visualize/visualizer.py:189]: Not valid.
The rendered examples are prototypes.

File: src/visualize/visualizer.py [hunk 7]
```diff
@@ -31,114 +46,124 @@ def __init__(self, net: ReProSeg, cfg: ReProSegConfig, log: Log):
...
-        with open(tensors_path, "wb") as f:
-            pickle.dump(self.tensors_per_prototype, f)
-
-    def render_prototype_activations(self):
-        self.log.info(f"Saving top {self.k} prototype activations to images...")
+            images = []
+            for idx in local_image_idxs:
+                image = pil_to_tensor(Image.open(image_paths[base_idx + idx]).convert("RGB"))
+                image = resize_image(image)
+                images.append(image)
```
[the remark this answers is not in the document]
Comment (PortikAbel) [reply src/visualize/visualizer.py:153]: @copilot apply changes based on [this feedback](https://github.com/PortikAbel/ReProSeg/pull/9#discussion_r3082922202)
